### PR TITLE
Add  ALM reg tests

### PR DESCRIPTION
**Pull request by marchdf**
## Summary

Add an ALM openfast reg tests. It's time. As much as we've been resisting this... It's critical enough and I recently broke something (#1385) that would have been caught if we had a reg test.

## Pull request type

Please check the type of change introduced:

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe):

## Checklist

The following is included:

- [ ] new unit-test(s)
- [x] new regression test(s)
- [ ] documentation for new capability

This PR was tested by running:

- the unit tests
  - [ ] on GPU 
  - [ ] on CPU 
- the regression tests
  - [ ] on GPU 
  - [ ] on CPU

### CODE REVIEW

**Comment by marchdf** [comment]
So far I have this.  Red = T0, first run, green = T1, first run, blue = T0, restart run, orange = T1, restart run. They all do the same thing, which is expected. And blue and orange start later (since they are a restart)

[turbines.pdf](https://github.com/user-attachments/files/18042561/turbines.pdf)

**Comment by marchdf** [comment]
And with this restart case I can confirm the breakage I saw in #1385 and that breakage is not in 3.2.0.

File: test/CMakeLists.txt [hunk 1]
```diff
@@ -323,6 +323,93 @@ if(AMR_WIND_ENABLE_W2A)
...
+      WORKING_DIRECTORY
+      "${ACT_UNIFORM_ALM_TEST_BINARY_DIR}"
+      COMMAND_ERROR_IS_FATAL ANY)
+    execute_process(
+      COMMAND bash -c "sed -i -e 's/1   WakeMod/0   WakeMod/g' ${RTEST_ACT_UNIFORM_ALM_TEST_BINARY_DIR}/${5MW_LAND_DIR}/${AERO_FILE}"
+      WORKING_DIRECTORY
+      "${ACT_UNIFORM_ALM_TEST_BINARY_DIR}"
+      COMMAND_ERROR_IS_FATAL ANY)
+    execute_process(
+      COMMAND bash -c "sed -i -e 's/True          YawDOF/False         YawDOF/g' ${RTEST_ACT_UNIFORM_ALM_TEST_BINARY_DIR}/${5MW_LAND_DIR}/${ELASTO_FILE}"
```
Comment (jrood-nrel) [inline test/CMakeLists.txt:402]: Unfortunately sed isn't portable.

Comment (marchdf) [reply test/CMakeLists.txt:402]: yeah... got another idea?

File: amr-wind/wind_energy/actuator/Actuator.cpp [hunk 2]
```diff
@@ -63,6 +64,7 @@ void Actuator::pre_init_actions()
         pp.query("type", type);
         pp1.query("type", type);
         AMREX_ALWAYS_ASSERT(!type.empty());
+        cnt_turbfastline += (type == "TurbineFastLine") ? 1 : 0;
```
Comment (lawrenceccheung) [inline amr-wind/wind_energy/actuator/Actuator.cpp:67]: I think it's the sum total of all OpenFAST (both `TurbFastLine` and `TurbFastDisk`) instances which must be equal to or less than the number of ranks.

**Review by lawrenceccheung — CHANGES_REQUESTED** [review]
Thanks @marchdf, this is great and something that has been sorely needed.  

I'm adding in a note that the input files were generated with the notebook https://gist.github.com/lawrenceccheung/a775e8eb5162e906da2a75ce67169139, in case we want to generate other reg test type problems in the future.  

Also, we should adopt a similar process of downloading the NREL5MW files from `OpenFAST/r-test` for the [benchmark problems](https://github.com/Exawind/exawind-benchmarks/tree/main/exawind/NREL_5MW_Turbine).  Or least, noting which commit they were downloaded from and documenting which changes to the openfast model were necessary for the benchmark problems to run.  I know it's for a separate problem, but we should be handling them consistently (Tagging @ndevelder on this).

Lawrence

File: amr-wind/wind_energy/actuator/Actuator.cpp (hunk 2 above)
Comment (marchdf) [reply amr-wind/wind_energy/actuator/Actuator.cpp:67]: good point. Fixed.

File: test/CMakeLists.txt (hunk 1 above)
Comment (marchdf) [reply test/CMakeLists.txt]: I actually thought of a better way. We carry a patch and `git apply` it. Added benefit of this is that if the patch doesn't apply due to file changes in openfast, we know about it immediately.

Comment (jrood-nrel) [reply test/CMakeLists.txt]: That's a great idea.

**Review by jrood-nrel — APPROVED** [review]

**Review by lawrenceccheung — APPROVED** [review]

**Pull request merged**